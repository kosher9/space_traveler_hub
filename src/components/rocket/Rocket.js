import React from 'react';

function Rocket(props) {
  const rocket = props;
  return (
    <>
      <div className="flex space-x-6 mb-10">
        <img className="w-52 h-52" src={rocket.flickr_images} alt="Rocket" />
        <div>
          <h1 className="text-xl">{rocket.name}</h1>
          <p className="mt-3 text-sm">{rocket.description}</p>
          <button
            type="button"
            className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded mt-3"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </>
  );
}

export default Rocket;
