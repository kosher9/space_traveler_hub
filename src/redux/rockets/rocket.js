import axios from 'axios';
import { ROCKETS_URL } from '../../url_config';

const initialState = [];
const GET_ROCKETS = 'GET_ROCKETS';

const getRocketsAction = (data) => {
  const rockets = data.map((rocket) => ({
    id: rocket.id,
    name: rocket.rocket_name,
    description: rocket.description,
    flickr_images: rocket.flickr_images,
  }));
  return {
    type: GET_ROCKETS,
    payload: rockets,
  };
};

export const getRockets = () => async (dispatch) => {
  await axios({
    method: 'get',
    url: ROCKETS_URL,
    responseType: 'json',
  }).then((res) => {
    dispatch(getRocketsAction(res.data));
  });
};

const rocketsReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_ROCKETS:
      return [
        ...action.payload,
      ];
    default:
      return state;
  }
};

export default rocketsReducer;
