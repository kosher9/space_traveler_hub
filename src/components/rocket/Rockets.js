import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRockets } from '../../redux/rockets/rocket';

function Rockets() {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    if (rockets.length === 0) {
      dispatch(getRockets());
    }
  }, []);
  return <div>Rockets</div>;
}

export default Rockets;
