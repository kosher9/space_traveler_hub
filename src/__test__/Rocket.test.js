import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rocket from '../components/rocket/Rocket';
import Rockets from '../components/rocket/Rockets';
import { bookingAction, getRocketsAction, cancelBookingAction } from '../redux/rockets/rocket';

describe('Rocket', () => {
  it('should render rocket component', () => {
    const rocket = render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
    expect(rocket).toMatchSnapshot();
  });

  it('should render rockets component', () => {
    const rockets = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(rockets).toMatchSnapshot();
  });

  it('should return an action with type GET_ROCKETS', () => {
    const data = [
      {
        id: 1,
        name: 'rocket1',
        description: 'description1',
        flickr_images: [],
      },
    ];
    const action = getRocketsAction(data);
    expect(action.type).toBe('GET_ROCKETS');
  });

  it('should return an action with type BOOKING', () => {
    const id = 1;
    const action = bookingAction(id);
    expect(action.type).toBe('BOOKING');
  });

  it('should should return an action with type CANCEL_BOOKING', () => {
    const id = 1;
    const action = cancelBookingAction(id);
    expect(action.type).toBe('CANCEL_BOOKING');
  });
});
