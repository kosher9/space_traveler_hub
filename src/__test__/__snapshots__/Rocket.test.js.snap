// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket should render rocket component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex space-x-6 mb-10"
      >
        <img
          alt="Rocket"
          class="w-52 h-52"
        />
        <div>
          <h1
            class="text-xl"
          />
          <p
            class="mt-3 text-sm"
          />
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded mt-3"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex space-x-6 mb-10"
    >
      <img
        alt="Rocket"
        class="w-52 h-52"
      />
      <div>
        <h1
          class="text-xl"
        />
        <p
          class="mt-3 text-sm"
        />
        <button
          class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded mt-3"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket should render rockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pl-24 pr-24 pt-12"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="pl-24 pr-24 pt-12"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
